scripts/unity/func_name_list.py

```diff
@@ -14,10 +14,10 @@ def func_names_from_header(in_file, out_file):
 
     with open(out_file, 'w') as f_out:
         # Regex match all function names in the header file
-        x = re.findall(r"(struct\s|\s?)(\w+\s)(\*?)(\w+?)(\(.*?\))(;)",
+        x = re.findall(r"^\s*(?:\w+[*\s]+)+(\w+?)\([\s\S]*?\);",
```

Comment:
Isn't `[\s\S]` just a verbose way of saying `.`? I see this used a few places in this PR.

```
\S

    Matches any character which is not a whitespace character. This is the opposite of \s
```

Using a character class and its complement in a set match (`[]`) should match any character, I guess, no?

Reply:
Good catch. 

There is a difference. `.` doesn't automatically match newline, but with the `S` flag it will. `[\s\S]` will match all the things `.` does, but it also includes newlines, even if the `S` flag isn't set. This can be useful if we only want to match newline in parts of the expression. I think I did this out of habit in this PR, but the `S` flag is actually set for this case (and probably several others), so I think you're right! I'll change it for the ones that have `S` set, and check if there are others that could use the `S` flag instead.